messaging/filter/service.go

```diff
@@ -484,6 +535,46 @@ func (s *Service) addAsymmetricFilter(keyAsym *ecdsa.PrivateKey, chatID string,
 	return &Filter{FilterID: id, Topic: whisper.BytesToTopic(topic)}, nil
 }
 
+func (s *Service) getMessages() []*Messages {
+	var response []*Messages
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	for chatID := range s.chats {
+		messages := s.getMessagesForChat(chatID)
+		if messages.Error != nil || len(messages.Messages) != 0 {
+			response = append(response, messages)
+		}
+	}
+
+	return response
+}
+
+func (s *Service) getMessagesForChat(chatID string) *Messages {
+	response := &Messages{}
+
+	response.Chat = s.chats[chatID]
```

Comment:
No mutex here? It's in `getMessages()` above.

Reply:
it's only called from `getMessages()` so it should be ok not to lock here as taken care from the calling function (`getMessages()`), is that correct?

Reply:
Correct but it's not obvious if another method would like to use this one. Not a big deal.